Assignment 2/src/main/java/com/nuclei/assignment/service/inputvalidation/InputValidationImpl.java

```diff
@@ -0,0 +1,101 @@
+package com.nuclei.assignment.service.inputvalidation;
+
+import com.nuclei.assignment.constants.ExceptionsConstantsUtils;
+import com.nuclei.assignment.constants.StringConstantsUtils;
+import com.nuclei.assignment.enums.Courses;
+import com.nuclei.assignment.exception.CustomException;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * User Validation Implementation containing method to validate different properties.
+ */
+public class InputValidationImpl implements InputValidation {
+  
+  /**
+   * The Logger.
+   */
+  private final Log logger = LogFactory.getLog(InputValidationImpl.class);
+  
+  @Override
+  public void validateFullName(final String name) throws CustomException {
+    if (name.isEmpty()) {
```

Comment:
Name it generic if you are using it in both fullname and address.

Reply:
done